[Term Entry] Python Math function: math.gcd() (#5042)
Codecademy/docs: content/python/concepts/math-module/terms/math-gcd/math-gcd.md

---
Title: 'math.gcd()' 
Description: 'Returns the Greatest Common Divisor (GCD) of two integers.' 
Subjects: 
  - 'Computer Science'
  - 'Data Science'
  - 'Discrete Math'
Tags:
  - 'Algorithms'
  - 'Functions'
  - 'Integers'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
---

The **`math.gcd()`** function in Python is used to calculate the Greatest Common Divisor (GCD) of two integers. 

## Syntax

```pseudo
math.gcd(x, y)
```

The `math.gcd()` function takes the following parameters:

- `x`: This is the first integer whose greatest common divisor is to be computed.
- `y`: This is the second integer whose greatest common divisor is to be computed.

The `math.gcd()` function returns an integer value that represents the greatest common divisor of the two integers `x` and `y`. If either `x` or `y` is zero, the GCD is the absolute value of the non-zero number. If both `x` and `y` are zero, the GCD is zero.

## Example

The example below uses the `math.gcd()` function to return the greatest common divisor of _54_ and _24_:

```py

> import math
> 
> print(math.gcd(54, 24))

Reviewer: ```suggestion
import math

# GCD of two integers.
print(math.gcd(54, 24))

# GCD of more than two integers.
print(math.gcd(54, 108, 216))

 # GCD of zero and non-zero integers
print(math.gcd(54, 0)) 
```

```

> The GCD of _54_ and _24_ is: 

Reviewer: ```suggestion
The output of the example code above is: 
```

```shell

> 6

Reviewer: ```suggestion
6
54
54
```

```

> 

Reviewer: ```suggestion

> Note: If `math.gcd()` function is called with no arguments or if all arguments are zero (`0`), it will return `0`.

```

> ---
> Title: 'math.gcd()' 
> Description: 'Calculates the Greatest Common Divisor (GCD) of two integers.' 
> Subjects: 
>   - 'Computer Science'
>   - 'Data Science'
>   - 'Discrete Math'
> Tags:
>   - 'Functions'
>   - 'Algorithms'
>   - 'Integers'
> CatalogContent: 
>   - 'learn-python-3'
>   - 'paths/computer-science'
> ---

Reviewer 2: ```suggestion
---
Title: 'math.gcd()' 
Description: 'Returns the Greatest Common Divisor (GCD) of two integers.' 
Subjects: 
  - 'Computer Science'
  - 'Data Science'
  - 'Discrete Math'
Tags:
  - 'Algorithms'
  - 'Functions'
  - 'Integers'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
---
```

Reviewer 2: Changed the description with a word due to plagiarism issues and arranged the tags in an alphabetically ascending order

The **`math.gcd()`** function in Python is used to calculate the Greatest Common Divisor (GCD) of two integers. 

## Syntax

```pseudo
math.gcd(x, y)
```

The `math.gcd()` function takes the following parameters:

> - `x` is the first integer.
> - `y` is the second integer.

Reviewer 2: ```suggestion
- `x`: This is the first integer whose greatest common divisor is to be computed.
- `y`: This is the second integer whose greatest common divisor is to be computed.
```

> The `math.gcd()` function returns an `integer` value, that represents the greatest common divisor of the two integers `x` and `y`. If either `x` or `y` is zero, the GCD is the absolute value of the non-zero number. If both `x` and `y` are zero, the GCD is zero.

Reviewer 2: ```suggestion
The `math.gcd()` function returns an integer value that represents the greatest common divisor of the two integers `x` and `y`. If either `x` or `y` is zero, the GCD is the absolute value of the non-zero number. If both `x` and `y` are zero, the GCD is zero.
```

## Example

> Use `math.gcd()` to return the greatest common divisor of `54` and `24`:

Reviewer 2: ```suggestion
The example below uses the `math.gcd()` function to return the greatest common divisor of _54_ and _24_:
```

Reviewer 2: Numbers can be represented in _italics_

```py
import math

print(math.gcd(54, 24))
```

The GCD of 54 and 24 is: 

```shell
6
```

## Codebyte Example

Run the following example that uses the `math.gcd()` function to understand how it works:

```codebyte/python
import math

x = 48
y = 18

result = math.gcd(x, y)

print(f"The GCD of {x} and {y} is {result}")
```

> 
> 

Reviewer 2: ```suggestion
```

Reviewer 2: extra lines can be removed
